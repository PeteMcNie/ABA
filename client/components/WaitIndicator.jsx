import React from 'react'
import { connect } from 'react-redux'

const WaitIndicator = props => {
  return props.wait
    ? <img className="waitIndicator" src='./images/new1.svg' />
    : null
}

function mapStateToProps (state) {
  return {
    wait: state.loading
  }
}

export default connect(mapStateToProps)(WaitIndicator)
