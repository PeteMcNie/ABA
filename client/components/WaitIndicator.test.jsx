import React from 'react'

import { renderWithRedux } from '../testing/utils'

import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'

import WaitIndicator from './WaitIndicator'

test('Wait indicator appears when content pending', async () => {
  const { container } = renderWithRedux(<WaitIndicator />, { initialState: { loading: true } })
  expect(container).toMatchSnapshot()
})

test('Wait indicator does not appear when content is not pending', async () => {
  const { container } = renderWithRedux(<WaitIndicator />, { initialState: { loading: false } })
  expect(container).toMatchSnapshot()
})
