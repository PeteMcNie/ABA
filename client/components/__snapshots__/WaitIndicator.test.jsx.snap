// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wait indicator appears when content pending 1`] = `
<div>
  <img
    class="waitIndicator"
    src="./images/new1.svg"
  />
</div>
`;

exports[`Wait indicator does not appear when content is not pending 1`] = `<div />`;
